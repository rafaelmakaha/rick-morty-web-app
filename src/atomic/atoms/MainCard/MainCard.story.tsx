import React from 'react'
import { Meta, Story } from '@storybook/react'
import MainCard, { IMainCard } from '.'

export default {
  title: 'Atoms/MainCard'
} as Meta

const Template: Story<IMainCard> = (args) => <MainCard {...args} />

export const Normal = Template.bind({})

const argTypes = {
  color: {
    control: 'color'
  }
}

const defaultArgs: IMainCard = {}

Normal.args = { ...defaultArgs }
Normal.argTypes = argTypes
