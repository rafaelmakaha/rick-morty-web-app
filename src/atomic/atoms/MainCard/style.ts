import styled from "styled-components";

export const Background = styled.div`
  display: flex;
  flex-direction: column;
  background-color: ${({ theme }) => theme.colors.lightGrey };
  left: 5%;
  min-height: 100vh;
  position: absolute;
  width: 90vw;
`
