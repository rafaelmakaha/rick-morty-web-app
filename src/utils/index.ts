export * from './assignTestId'
export * from './withProviders'
