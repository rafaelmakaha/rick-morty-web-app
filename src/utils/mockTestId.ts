type componentType = 'div' | 'text' | 'input' | 'button' | 'img'

export const assignTestId = (
  componentType: componentType,
  componentId: string
) =>
  process.env.NODE_ENV !== 'production'
    ? {
        'data-testid': `${componentType}_${componentId}`
      }
    : {}

export const mockTestID = (
  componentType: componentType,
  componentId: string
): string => assignTestId(componentType, componentId)['data-testid'] ?? ''
