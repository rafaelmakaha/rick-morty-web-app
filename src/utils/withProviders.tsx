import React from 'react'

import { ThemeProvider } from 'styled-components'

import { theme } from '../tokens'

export const StorybookProviders: React.FC = ({ children }) => (
  <ThemeProvider theme={theme}>{children}</ThemeProvider>
)

export const providerDecorator = (Story: any) => (
  <StorybookProviders>
    <Story />
  </StorybookProviders>
)
