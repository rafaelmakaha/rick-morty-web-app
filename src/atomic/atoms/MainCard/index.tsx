import React from "react";
import { Background } from "./style";

export interface IMainCard {
  testID?: string
  color?: string
  children?: React.ReactNode
}

const MainCard: React.FC<IMainCard> = ({ testID = 'MainCard', color, children }) => (
  <Background color={color}>{children}</Background>
);

export default MainCard;
