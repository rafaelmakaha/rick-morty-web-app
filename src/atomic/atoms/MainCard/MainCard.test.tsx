import { cleanup, render } from "@testing-library/react";
import React from "react";
import MainCard, { IMainCard } from ".";
import { mockTestID } from "../../../utils";

describe('Atoms/MainCard', () => {
    const props: IMainCard = {
        testID: 'MainCard',
        color: 'red',
        children: 'Some Children'
    }

    const renderComponent = () => render(<MainCard {...props} />)

    let component = renderComponent()

    beforeEach(() => {
        jest.clearAllMocks()
        cleanup()
        component = renderComponent()
    })

    const el_container = mockTestID('div', props.testID!)

    it(`should render component #${el_container}`, () => {
        const sut = component.findByTestId(el_container)
        expect(sut).toBeTruthy()
    })

    it('should render snapshot', () => {
        expect(component.container.firstChild).toMatchSnapshot()
    })

})
