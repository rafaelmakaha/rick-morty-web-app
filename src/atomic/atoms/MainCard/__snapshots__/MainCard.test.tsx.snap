// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Atoms/MainCard should render snapshot 1`] = `
<div
  class="style__Background-sc-1o95tzs-0 jeHnGC"
  color="red"
>
  Some Children
</div>
`;
